packages/liveblocks-node/src/client.ts

```diff
@@ -896,12 +898,25 @@ export class Liveblocks {
      * }
      * ```
      */
-    query?: string;
+    query?:
+      | string
+      | {
+          metadata?: Partial<QueryMetadata<TThreadMetadata>>;
+        };
   }): Promise<{ data: ThreadData[] }> {
     const { roomId } = params;
 
+    let query: string | undefined;
+
+    if (typeof params.query === "string") {
+      query = params.query;
+    } else if (typeof params.query === "object") {
+      // @ts-expect-error: FIXME typing
```

Comment:
struggling to fix this TS error 🤔

```
Argument of type '{ metadata?: Partial<QueryMetadata<TThreadMetadata>> | undefined; }' is not assignable to parameter of type '{ [key: string]: FilterValue | { [key: string]: FilterValue; }; }'.
  Property 'metadata' is incompatible with index signature.
    Type 'Partial<QueryMetadata<TThreadMetadata>>' is not assignable to type 'FilterValue | { [key: string]: FilterValue; }'.
      Type 'Partial<QueryMetadata<TThreadMetadata>>' is not assignable to type '{ [key: string]: FilterValue; }'.
        Type 'QueryMetadata<TThreadMetadata>[P] | undefined' is not assignable to type 'FilterValue'.
          Type 'undefined' is not assignable to type 'FilterValue'.ts(2345)```

Reply:
I think this will go away once you allow explicit-`undefined`s, see my earlier comment above.

Reply:
That was it, thank you!